### PR TITLE
Add Tests for models/tx

### PR DESCRIPTION
**Pull request by hectorchu**

### CODE REVIEW

File: test/framework/include/test_framework/TxBuilder.h [hunk 1]
```diff
@@ -110,7 +110,7 @@ struct TxBuilder
         Kernel kernel = Kernel::CreatePegOut(amount, fee, std::move(ltc_address), std::move(excess_commitment), std::move(signature));
 
         m_kernels.push_back(std::move(kernel));
-        m_amount -= amount;
+        m_amount -= amount + fee;
```
Comment (DavidBurkett) [inline test/framework/include/test_framework/TxBuilder.h:113]: Good catch!

File: test/tests/models/tx/Test_Input.cpp [hunk 2]
```diff
@@ -1 +1,93 @@
...
\ No newline at end of file
+#include <catch.hpp>
+
+#include <mw/crypto/Crypto.h>
+#include <mw/crypto/Random.h>
+#include <mw/models/tx/Input.h>
+
+TEST_CASE("Plain Tx Input")
+{
+    Commitment commit(Random::CSPRNG<33>().GetBigInt());
```
Comment (DavidBurkett) [inline test/tests/models/tx/Test_Input.cpp:10]: If you switched this to use Commitment::FromHex instead, then you could assert that the serialized input is an exact byte array, which seems more ideal than just recreating the serialization logic in the test.

This is probably good for now though. We can just make a note to go back and simplify this later.

btw, I also took the same route with kernel tests, so I completely understand why you implemented it this way. It was a poor choice on my part.

File: test/tests/models/tx/Test_PegOutCoin.cpp [hunk 3]
```diff
@@ -1 +1,46 @@
...
\ No newline at end of file
+#include <catch.hpp>
+
+#include <mw/crypto/Crypto.h>
+#include <mw/models/tx/PegOutCoin.h>
+
+TEST_CASE("Tx Peg-Out Coin")
+{
+    uint64_t amount = 123;
+    Bech32Address address = Bech32Address::FromString("bc1qc7slrfxkknqcq2jevvvkdgvrt8080852dfjewde450xdlk4ugp7szw5tk9");
```
Comment (mosadialiou) [inline test/tests/models/tx/Test_PegOutCoin.cpp]: bc1qc7slrfxkknqcq2jevvvkdgvrt8080852dfjewde450xdlk4ugp7szw5tk9 is not valid litecoin bech32 address instead it looks like a bitcoin bech32 address. So is it OK to use a bitcoin bech32 address here?

Comment (DavidBurkett) [reply test/tests/models/tx/Test_PegOutCoin.cpp]: Yes, it's perfectly fine to use this address. Bech32 addresses have something called an "HRP" (Human readable part) that we set to different values for each coin *by convention* only. So the hrp of this string won't be the same one used by ltc wallets, but that's ignored by the protocol anyway. The important part is the actual pubkey that's encoded in it.

**Review by DavidBurkett — APPROVED** [review]

File: test/tests/models/tx/Test_PegOutCoin.cpp (hunk 3 above)
Comment (mosadialiou) [reply test/tests/models/tx/Test_PegOutCoin.cpp]: ok sounds good.
Even for consistency? it's for litecoin so shouldn't we use litecoin address?

Comment (DavidBurkett) [reply test/tests/models/tx/Test_PegOutCoin.cpp]: Sure, this one is just a bech32 address that's used as an example bech32 address in multiple places like this: https://en.bitcoin.it/wiki/Bech32

Do you happen to know a ltc bech32 address we can change it to? If not, I can generate one.

Comment (mosadialiou) [reply test/tests/models/tx/Test_PegOutCoin.cpp]: This one tltc1qh50sy0823vxn4l9zk2820w4fuj0q4fgv48ma6c ?

File: test/tests/models/tx/Test_Kernel.cpp [hunk 4]
```diff
@@ -276,4 +276,84 @@ TEST_CASE("Height-Locked")
...
+    //
+    // JSON
+    //
+    {
+        Json json(kernel.ToJSON());
+        REQUIRE(json.GetKeys() == std::vector<std::string>({ "excess", "fee", "signature", "type" }));
+        REQUIRE(json.GetRequired<Commitment>("excess") == excess);
+        REQUIRE(json.GetRequired<uint64_t>("fee") == fee);
+        REQUIRE(json.GetRequired<Signature>("signature") == signature);
+        REQUIRE(json.GetRequired<std::string>("type") == "UNKNOWN");
```
Comment (mosadialiou) [inline test/tests/models/tx/Test_Kernel.cpp:326]: Did we already test other types in KernelType?

Comment (DavidBurkett) [reply test/tests/models/tx/Test_Kernel.cpp:326]: Yes. You can see the whole file with the changes here: https://github.com/ltc-mweb/libmw/blob/ab939d6cd1fe3f523f846b4ccbe17960d4e88552/test/tests/models/tx/Test_Kernel.cpp

**Pull request merged**